tests/ErrorHandlerTest.php

```diff
@@ -358,6 +278,50 @@ public function testHandleExceptionWithThrowingPreviousExceptionHandler(): void
         } finally {
             restore_error_handler();
             restore_exception_handler();
+
+            $this->assertSame(2, $listenerCalled);
+            $this->assertSame($exception1, $captured1);
+            $this->assertSame($exception2, $captured2);
         }
     }
+
+    /**
+     * @dataProvider callableProvider
+     */
+    public function testAddListener(callable $listener): void
```

Comment:
This test doesn't seems solid, if for some reason the exception is not thrown from `handleException` the test will probably just say that no assertion has been made

Reply:
Good catch!